### PR TITLE
 i18n(es): update the TypeScript guide.

### PR DESCRIPTION
**Pull request by diegognt**
#### What kind of changes does this PR include?
- New or updated content
- Translated content

#### Description
 - Keeping up to date the page with the English content
 - Rephrasing several paragraphs on the page for more clear context

### CODE REVIEW

File: src/pages/es/guides/typescript.md [hunk 1]
```diff
@@ -11,26 +11,15 @@ Astro no realiza ninguna verificación de tipos por sí mismo. La verificación
 
 ## Configuración
 
-Se **recomienda encarecidamente** que crees un archivo `tsconfig.json` para que las herramientas como Astro y VSCode sepan cómo interpretar tu proyecto. Algunas funcionalidades (como las importaciones de paquetes npm) no son totalmente compatibles con TypeScript a menos que crees un archivo `tsconfig.json`.
+[] incluye un archivo `tsconfig.json` en tu proyecto. Incluso si tu no escribes código TypeScript, este archivo es importante pues así, herramientas como Astro y VS Code saben como entender tu proyecto. Algunas características(como importación de archivos npm) no están totalmente soportadas en el editor sin un archivo `tsconfig.json`. Si instalas Astro manualmente, asegurate de create este archivo por tú mismo.
```
Comment (fredoist) [inline src/pages/es/guides/typescript.md:14]:
```suggestion
Los *starter projects* de Astro incluyen un archivo `tsconfig.json` en tu proyecto. Incluso si no escribes código TypeScript, este archivo es importante para que herramientas como Astro y VS Code sepan cómo interpretar tu proyecto. Algunas características (como importaciones de paquetes npm) no están totalmente soportadas en el editor sin un archivo `tsconfig.json`. Si instalas Astro manualmente, asegúrate de crear este archivo por ti mismo.
```

File: src/pages/es/guides/typescript.md [hunk 2]
```diff
@@ -11,26 +11,15 @@ Astro no realiza ninguna verificación de tipos por sí mismo. La verificación
...
 
-Algunas opciones de configuración de TypeScript requieren atención especial en Astro. A continuación te recomendamos un archivo `tsconfig.json` básico, que puedes copiar y pegar en tu proyecto. Cada [plantilla en astro.new](https://astro.new/) incluye este archivo `tsconfig.json` por defecto.
+Tres plantillas extensibles del archivo `tsconfig.json` están incluidas con Astro: `base`, `strict` y `strictest`. La plantilla `base` habilita el soporte para para características modernas de JavaScript y también es la base de las otras plantillas. Nosotros recomendamos usar las plantillas `strict` o `strictest` si planeas escribir TypeScript en tu proyecto. Puedes ver y comparar las tres plantillas en [astro/tsconfigs/](https://github.com/withastro/astro/blob/main/packages/astro/tsconfigs/).
```
Comment (fredoist) [inline src/pages/es/guides/typescript.md:16]:
```suggestion
Tres plantillas extensibles del archivo `tsconfig.json` están incluidas con Astro: `base`, `strict` y `strictest`. La plantilla `base` habilita el soporte para características modernas de JavaScript y también es usada como base para las otras plantillas. Recomendamos usar las plantillas `strict` o `strictest` si planeas escribir TypeScript en tu proyecto. Puedes ver y comparar las tres plantillas en [astro/tsconfigs/](https://github.com/withastro/astro/blob/main/packages/astro/tsconfigs/).
```

**Review by fredoist — CHANGES_REQUESTED** [review]
Well done @diegognt 🤝 
I left a couple suggestions, please check them out and feel free to add or modify anything.

File: src/pages/es/guides/typescript.md (hunk 1 above)
Comment (diegognt) [reply src/pages/es/guides/typescript.md]: Thanks for the recommendations. I will implement the suggested updates.

P.S. That `[]` my dear Lord...

File: src/pages/es/guides/typescript.md (hunk 2 above)
Comment (diegognt) [reply src/pages/es/guides/typescript.md]: Thanks for the recommendations. I will implement the suggested updates.

**Review by agustinmulet — APPROVED** [review]
LGTM! ✅

**Pull request merged**